src/jit/importer.cpp

```diff
@@ -12853,15 +12854,22 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                             lvaSetStruct(lclNum, resolvedToken.hClass, true /* unsafe value cls check */);
                         }
 
-                        // Append a tree to zero-out the temp
-                        newObjThisPtr = gtNewLclvNode(lclNum, lvaTable[lclNum].TypeGet());
-
-                        newObjThisPtr = gtNewBlkOpNode(newObjThisPtr,    // Dest
-                                                       gtNewIconNode(0), // Value
-                                                       size,             // Size
-                                                       false,            // isVolatile
-                                                       false);           // not copyBlock
-                        impAppendTree(newObjThisPtr, (unsigned)CHECK_SPILL_NONE, impCurStmtOffs);
+                        // Structs with GC pointer fields are fully zero-initialized in the prolog if compInitMem is
+                        // true.
+                        // Therefore, we don't need to insert zero-initialization here if this block is not in a loop.
+                        if (!info.compInitMem || !containsGCPtr || ((block->bbFlags & BBF_BACKWARD_JUMP) != 0) ||
```

Comment:
You can refactor this expression into its own method since your change uses it twice.

`!info.compInitMem ||  ((block->bbFlags & BBF_BACKWARD_JUMP) != 0)`

Reply:
@briansull I factored the expression into its own method. PTAL.